service/worker/deployment/deployment_client.go

```diff
@@ -367,8 +373,7 @@ func (d *DeploymentClientImpl) ListDeployments(
 		deployments = append(deployments, deploymentListInfo)
```

Comment:
nit: while we're here, let's change this loop to `make(..., len(persistenceResp.Executions)` and assign by index?